### DISCUSSION TITLE
Add Support for Customizable OpenID Profile Fields via Environment Variable

### DISCUSSION THREAD

**Discussion by geodanchev** [danny-avila/LibreChat#4354] [Feature Requests & Suggestions]
I would like to propose adding support for customizable OpenID profile fields in the user model using an environment variable.

**Overview:**
This feature would introduce a new environment variable, `OPENID_CUSTOM_DATA`, that allows developers to define which fields from the OpenID profile should be dynamically fetched and stored in the user model. The fields specified in this variable will be fetched from the OpenID provider (e.g., Microsoft Graph API) and saved in a new, flexible `customOpenIdData `field in the user schema.

**Key Benefits:**

1. **Flexibility:** Developers can define custom fields to be fetched from the OpenID provider at runtime, such as department, title, or any other available field.
2.  **Scalability:** The `customOpenIdData `field in the user schema is a Map type, allowing it to easily store key-value pairs of custom profile data.
3. **Backward Compatibility:** This change is optional and will only be applied when the `OPENID_CUSTOM_DATA `environment variable is set. If the variable is not defined, the system will continue to function as it does currently.

**Suggested Implementation:**

 **Environment Variable:** Introduce the `OPENID_CUSTOM_DATA`variable to define custom profile fields (space-separated) to be fetched from the OpenID provider.

- Example: `OPENID_CUSTOM_DATA="department title"`

 **User Schema:** Extend the user schema with a `customOpenIdData` field of type `Map<String, String>`, allowing flexible storage of the fetched data.

 **Data Population:** Upon user login, fetch the fields specified in `OPENID_CUSTOM_DATA `from the OpenID provider, and store them in the `customOpenIdData` field for each user.

**Example:**

If `OPENID_CUSTOM_DATA="department title"` is set, the following data could be stored in MongoDB for a user:

```
{
  "customOpenIdData": {
    "department": "AI Process Optimization",
    "title": "Senior Developer"
  }
}
```

> This feature would enable developers to extend and customize user profiles with minimal effort, without impacting existing functionality. It provides a simple, scalable way to manage additional profile information from OpenID providers. 

In my case this feature will help me make useful statistics for example : how many users per department are using the app and provide department help and use specific use case scenarios, to help users better understand how they can benefit from AI.

**Comment by danny-avila** [comment]
Great idea! I'm all for it

**Reply by geodanchev** [reply]
Great! I've pushed the changes [here](https://github.com/danny-avila/LibreChat/pull/4362)

**Comment by rubentalstra** [comment]
request a pull request with a more dynamic approach  #5612
